src/cpp/src/whisper/whisper.cpp

```diff
@@ -129,22 +130,49 @@ int64_t decode_with_past(ov::Tensor& encoder_hidden_state,
     return output_token;
 }
 
-std::vector<int64_t> prepare_input_ids(const ov::genai::WhisperGenerationConfig& config) {
-    if (config.is_multilingual) {
-        return std::vector<int64_t>{config.decoder_start_token_id,
-                                    config.language_token_id,
-                                    config.transcribe_token_id,
-                                    config.no_timestamps_token_id};
+int64_t detect_language(ov::Tensor& encoder_hidden_state,
+                        ov::InferRequest decoder,
+                        const ov::genai::WhisperGenerationConfig& config) {
+    std::vector<int64_t> input_ids{config.decoder_start_token_id};
+    int64_t output_token = decode(encoder_hidden_state, decoder, input_ids, config, false);
```

Comment:
Currently with language autodetection 2 decoders (without past) calls used. I believe this could be done with 1 call. I'll add todo for that.